tbmalt/common/maths/interpolation.py

```diff
@@ -205,33 +228,33 @@ class PolyInterpU(Feed):
     polynomial values smoothly converge to zero at the boundary.
 
     Arguments:
-        xx: Grid points for interpolation, 1D Tensor.
-        yy: Values to be interpolated at each grid point.
+        x: Grid points for interpolation, 1D Tensor.
+        y: Values to be interpolated at each grid point. Note that this must
+            be a `torch.nn.Parameter` rather than a standard torch `Tensor`.
         tail: Distance to smooth the tail.
         delta_r: Delta distance for 1st, 2nd derivative.
         n_interp: Number of total interpolation grid points.
         n_interp_r: Number of right side interpolation grid points.
 
     Attributes:
-        _xx: Grid points for interpolation, 1D Tensor.
-        yy: Values to be interpolated at each grid point.
         delta_r: Delta distance for 1st, 2nd derivative.
         tail: Distance to smooth the tail.
         n_interp: Number of total interpolation grid points.
         n_interp_r: Number of right side interpolation grid points.
-        grid_step: Distance between each gird points.
 
     Notes:
         The `PolyInterpU` class, which is taken from the DFTB+, assumes a
-        uniform grid. Here, the yy and xx arguments are the values to be
+        uniform grid. Here, the y and x arguments are the values to be
         interpolated and their associated grid points respectively. The tail
         end of the spline is smoothed to zero, meaning that extrapolated
         points will rapidly, but smoothly, decay to zero.
 
     Examples:
+        >>> import torch
+        >>> from torch.nn import Parameter
         >>> import matplotlib.pyplot as plt
         >>> x = torch.linspace(0, 2. * torch.pi, 100)
-        >>> y = torch.sin(x)
+        >>> y = Parameter(torch.sin(x), requires_grad=False)
         >>> poly = PolyInterpU(x, y, n_interp=8, n_interp_r=4)
         >>> new_x = torch.rand(10) * 2. * torch.pi
         >>> new_y = poly(new_x)
```

Comment:
Hey Adam, seems the forward method should be used.
`new_y = poly.forward(new_x)`